examples/berad/pkg/state-transition/state_processor.go

```diff
@@ -110,6 +111,7 @@ func NewStateProcessor[
 	],
 	ExecutionPayloadHeaderT ExecutionPayloadHeader,
 	ForkT interface {
+		Fork[ForkT]
 		New(common.Version, common.Version, math.Epoch) ForkT
```

Comment:
add the `New` function in the `Fork` interface you defined in types.go in this package.